src/util/queryNodeMatchers.ts

```diff
@@ -0,0 +1,147 @@
+import { Position, Range } from "vscode";
+import { SyntaxNode, Query, Language, QueryCapture } from "web-tree-sitter";
+import {
+  NodeMatcher,
+  NodeMatcherValue,
+  SelectionExtractor,
+  SelectionWithEditor,
+} from "../typings/Types";
+import { simpleSelectionExtractor } from "./nodeSelectors";
+
+export function defaultMatcher(
+  scopeType: string,
+  searchScopePresent: boolean,
+  scopeQuery: string,
+  selector: SelectionExtractor = simpleSelectionExtractor
+): NodeMatcher {
+  let query: Query;
+
+  return (
+    selection: SelectionWithEditor,
+    node: SyntaxNode,
+    siblings: boolean | undefined
+  ): NodeMatcherValue[] | null => {
+    if (!query) {
+      query = getQuery(node, scopeQuery);
```

Comment:
I would still prefer to pass in the language rather than the node, to make it clear we don't actually care about the node itself